CMakeLists.txt

```diff
@@ -0,0 +1,135 @@
+#
+# CMakeLists file for OpenSSL
+# Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+
+cmake_minimum_required(VERSION 3.14)
+
+project(OpenSSL LANGUAGES C CXX
+        DESCRIPTION "Open-source SSL and Cryptographic library"
+        HOMEPAGE_URL "https://github.com/openssl/openssl")
+
```

Comment:
Can do also search for NASM_EXECUTABLE for Windows users?
Lets expect users to set cmake cache variable NASM_ROOT.
Adding stuff to the system PATH is well, considered a bad practice.
Can we add something like this?
```suggestion
# If we compile on Windows or cross-compile for Windows
if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Windows" OR "${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
    find_program(NASM_EXECUTABLE nasm REQUIRED
        HINTS "$ENV{NASM_ROOT}" "${NASM_ROOT}"
        PATH_SUFFIXES bin)
endif()
```
And then pass this to Configure script? (Configure discovers nasm executable by some environmental variable or does it needs to have it in the PATH by all means?)
Can't really validate version here without some ugly output parsing by some regexes... well I guess just don't do this or correct me if I am wrong and this is really important to check.

Reply:
> Adding stuff to the system PATH is well, considered a bad practice.

In part because cmake then tries to compile against libraries that are next to it? :P

Reply:
That is a circular dependency.
CMake uses system PATH for discoveries knowing that it is not very good but to provide maximal support for projects that do use PATH, and developers in those projects use PATH because CMake supports it. Somebody needs to break the loop 😂